gap/Algebroids.gi

```diff
@@ -57,7 +57,46 @@ InstallMethod( SetOfObjects,
         "for an algebroid",
         [ IsAlgebroid and HasUnderlyingQuiver ],
         
-  A -> List( Vertices( UnderlyingQuiver( A ) ), o -> A.(String( o ) ) ) );
+  A -> List( Vertices( UnderlyingQuiver( A ) ), o -> A.( String( o ) ) ) );
+
+##
+InstallMethod( AssignSetOfObjects,
+        [ IsAlgebroid and HasUnderlyingQuiver, IsString ],
+  function( A, label )
+    local names, objects, func;
+    
+    names := List( Vertices( UnderlyingQuiver( A ) ), String );
+    
+    objects := SetOfObjects( A );
+    
+    func := function( name, o )
+              
+              if Int( name ) <> fail and label = "" then
+                  Error( "The second argument should be a non-empty string" );
+              fi;
+              
+              name := Concatenation( label, ReplacedString( name, "-", "m" ) );
+              
+              MakeReadWriteGlobal( name );
+              
+              DeclareSynonym( name, o );
+              
+              return 1;
+              
+            end;
+            
+    ListN( names, objects, func );
+    
+end );
+
+##
+InstallOtherMethod( AssignSetOfObjects,
+        [ IsAlgebroid and HasUnderlyingQuiver ],
+  function( A )
+  
+    AssignSetOfObjects( A, "" );
```

Comment:
What does this line mean?

Reply:
if the method is called with two arguments then concatenate the second argument with the labels of the vertices. If the method is called with one argument, i.e., with algebroid, then just use the labels of  the vertices. For example, if the algebroid is "a:1->2", then I would apply
`AssignSetOfObjects( A, "P_" );` But if the algebroid is `"a:T_1->T_2"`, then I would just apply 
`AssignSetOfObjects( A );` which calles `AssignSetOfObjects( A, "" );`